src/container/srv_target.c

```diff
@@ -2537,3 +2660,114 @@ ds_cont_ec_timestamp_update(struct ds_cont_child *cont)
 {
 	cont->sc_ec_update_timestamp = d_hlc_get();
 }
+
+void
+ds_cont_child_insert_dtx(struct ds_cont_child *cont_child, struct dtx_leader_handle *new_dlh)
+{
+	struct dtx_leader_handle *dlh = NULL;
+
+	/* Open DTXs should be ordered in the list, and most likely the new inserted DTX
+	 * should be inserted at the head of the list.
+	 */
+	d_list_for_each_entry(dlh, &cont_child->sc_open_dtx_list, dlh_link_list) {
+		if (dlh->dlh_handle.dth_epoch < new_dlh->dlh_handle.dth_epoch) {
+			d_list_add_tail(&new_dlh->dlh_link_list, &dlh->dlh_link_list);
```

Comment:
so the sc_open_dtx_list is ordered by epoch order (last is the highest epoch)

Reply:
yes, since it is per xstream, and these dtx is added to the list during start, where it use current HLC (naturally ordered by time anyway), and the 1st one is the oldest one anyway.